### ISSUE TITLE
Consider using pytest-mypy-plugins for type tests

### ISSUE THREAD

**Issue by sobolevn** [dbrgn/result#41]
I have notied that you have this file: https://github.com/dbrgn/result/blob/master/result/typetests.py

It might be a good idea to switch to https://github.com/typeddjango/pytest-mypy-plugins if you need more features like:
- Testing errors
- Testing different `mypy` setups

That's how we test our `Result`: https://github.com/dry-python/returns/tree/master/typesafety/test_result

**Comment by francium** [comment]
@sobolevn Hey, I had a quick look at this stuff. I'd prefix this by saying I don't do much Python day to day, only as a 'hobby' from time to time ;)

~~After going it, I'm a bit unsure of the benefits as opposed to just the current setup which runs mypy over the python file we've got (typetest.py). It's a bit ad-hoc for sure and we've have to separately include it in the CI testing setup as opposed to pytest-mypy-plugins which automatically picks up the yaml test file and runs it without any extra config.~~

~~Is there any additional benefits we would gain from using pytest-mypy-plugins over running mypy over the file like we've got currently~~

So after writing this all out and re-reading your comment,
I guess these are the benefits over just using mypy directly,
- Testing errors -- mypy will just fail, but with the plugin we can assert things don't type check in expected cases
- Testing different mypy setups -- different configurations? (`mypy_config`)

Hmm, after writing this I can definitely see some benefits. I think I'm inclined to switch over, unless @dbrgn you've got some concerns as well?
